### PR TITLE
Prepare for version 5.0 [wip]

### PR DESCRIPTION
**Pull request by srknzl**
This PR aims to release 5.0 version of this project that'll make it support all 5.0+ versions of Hazelcast. The original request ticket can be found [here](https://hazelcast.atlassian.net/browse/SUP-420).

The version upgrades are done by taking these into account: 

- latest where possible
- There should be a CVE associated
- Should work with java 1.8
- SHould work together

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -103,9 +104,19 @@
     <repositories>
         <repository>
             <id>snapshot-repository</id>
-            <name>Maven2 Snapshot Repository</name>
+            <name>Hazelcast Snapshot Repository</name>
```
Comment (frant-hartm) [inline pom.xml:107]: This is not Hazelcast repository, it's maven central repo for snapshot

File: pom.xml [hunk 2]
```diff
@@ -103,9 +104,19 @@
...
+        <repository>
+            <id>main-repo</id>
+            <name>Main Repository</name>
+            <url>https://repo1.maven.org/maven2</url>
+        </repository>
+        <repository>
+            <id>backup-repo</id>
+            <name>Backup Repository</name>
+            <url>https://repo2.maven.org/maven2/</url>
+        </repository>
```
Comment (frant-hartm) [inline pom.xml:119]: Why did you add these?

File: pom.xml [hunk 3]
```diff
@@ -31,17 +31,18 @@
...
         <junit.version>4.13.1</junit.version>
 
         <jsp.api.version>2.2.1</jsp.api.version>
         <servlet.api.version>3.1.0</servlet.api.version>
-        <org.springframework.version>4.3.9.RELEASE</org.springframework.version>
-        <spring-security.version>4.2.3.RELEASE</spring-security.version>
-        <jetty.maven.plugin.version>8.1.2.v20120308</jetty.maven.plugin.version>
+        <org.springframework.version>5.3.34</org.springframework.version>
+        <spring-security.version>5.8.12</spring-security.version>
+        <jetty.maven.plugin.version>9.4.54.v20240208</jetty.maven.plugin.version>
```
Comment (frant-hartm) [inline pom.xml:43]: We need to figure out what versions of Jetty and Tomcat we should support with this new version. This should be based on what is maintained on Jetty / Tomcat side and what our customers currently use.

File: README.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ We need to have all these sessions backed up somewhere if we do not want to lose
 
 The following are required before you can enable Hazelcast Session Clustering:
 
--   Target application or web server should support Java 1.6 or higher.
+-   Target application or web server should support Java 1.8 or higher.
```
Comment (frant-hartm) [inline README.md:34]: This actually depends on Hazelcast version
5.0 - 5.2 - 1.8
5.3 - 11
5.4 - 17+

**Review by frant-hartm — CHANGES_REQUESTED** [review]
How are we going to deal with the javax/jakarta package name issue?

File: pom.xml (hunk 1 above)
Comment (srknzl) [reply pom.xml:107]: This came from https://github.com/hazelcast/hazelcast-wm/commit/9fb91a4ae7a939c4db49d2d87a041dd71c33329f. So @Fly-Style, can you weigh in?

File: pom.xml (hunk 2 above)
Comment (srknzl) [reply pom.xml:119]: This came from https://github.com/hazelcast/hazelcast-wm/commit/9fb91a4ae7a939c4db49d2d87a041dd71c33329f. So @Fly-Style, can you weigh in?

Comment (srknzl) [reply pom.xml:119]: I thought they were required, but if not, let me delete them

File: pom.xml (hunk 3 above)
Comment (srknzl) [reply pom.xml:43]: This is just a test dependency, you mean for better testing (better means close to their environment) right ?
